cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreImplForceInsertTest.java

```diff
@@ -251,6 +259,44 @@ public void forceInsert_sameLengthOfPath_remoteRevisionWins() throws Exception {
         Assert.assertTrue(Arrays.equals(bodyOne.asBytes(), obj.getBody().asBytes()));
     }
 
+    public void forceInsert_sameRevisionTwice() throws Exception {
+        final String sql = "SELECT docs.docid, revid from revs join docs on revs.doc_id  == docs.doc_id where docs.docid = ? and revs.revid = ? ";
+
+
+        final DocumentRevision rev = createDbObject("1-rev", bodyOne);
+        EventSubscriber eventSubscriber = new EventSubscriber();
+        datastore.getEventBus().register(eventSubscriber);
+        datastore.forceInsert(rev, "1-rev");
+        Assert.assertThat(datastore.getDocumentCount(), is(1));
+        Assert.assertThat(eventSubscriber.eventCount, is(1));
+        datastore.getEventBus().unregister(eventSubscriber);
```

Comment:
An easier way to do this is to skip these two lines (the `unregister` and the `new`) and just check that `eventCount` is still 1